### PR TITLE
Code and test for issue:Add methods for converting iterators to primi…

### PR DESCRIPTION
**Pull request by sambohrer**
…tive iterators #2666

### CODE REVIEW

File: guava/src/com/google/common/collect/Iterators.java [hunk 1]
```diff
@@ -1303,4 +1304,38 @@ public ConcatenatedIterator(Iterator<? extends Iterator<? extends T>> iterators)
...
+  }
+  
+  /**
+   * Converts a Long iterator into a corresponding primitive iterator
+   * @param iterator iterator to convert
+   * @return converted iterator
+   */
+  public static PrimitiveIterator.OfLong toLongIterator(Iterator<Long> iterator) {
+	  checkNotNull(iterator);
+	  List<Long> list = Lists.newArrayList(iterator);
```
Comment (fenik17) [inline guava/src/com/google/common/collect/Iterators.java]: Why need collect iterator to list? May be better create stream from Iterable.spliterator() :
```java
        Iterable<Long> longs = () -> iterator;
        return StreamSupport.stream(longs.spliterator(), false)
                .mapToLong(n -> n)
                .iterator();
```

**Comment by orionll** [comment]
There is soo much overhead in this implementation...

**Comment by fenik17** [comment]
And streams unavailable: guava support java 6+ ..

**Comment by orionll** [comment]
Next Guava will require Java 8

File: guava/src/com/google/common/collect/Iterators.java [hunk 2]
```diff
@@ -1303,4 +1304,38 @@ public ConcatenatedIterator(Iterator<? extends Iterator<? extends T>> iterators)
...
   }
+  
+  
+  /**
+   * Converts an Integer iterator into a corresponding primitive iterator
+   * @param iterator iterator to convert
+   * @return converted iterator
+   */
+  public static PrimitiveIterator.OfInt toIntIterator(Iterator<Integer> iterator) {
+	  checkNotNull(iterator);
```
Comment (liach) [inline guava/src/com/google/common/collect/Iterators.java]: Uh, you should not use tabs.

Comment (liach) [inline guava/src/com/google/common/collect/Iterators.java]: You should add a check:
```java
if (iterator instanceof PrimitiveIterator.OfInt) {
  return (PrimitiveIterator.OfInt) iterator;
}
```
And add test cases for these.

File: guava/src/com/google/common/collect/Iterators.java (hunk 1 above)
Comment (liach) [inline guava/src/com/google/common/collect/Iterators.java]: Ideal code:
```java
  public static PrimitiveIterator.OfLong toLongIterator(Iterator<Long> iterator) {
    checkNotNull(iterator);
    if (iterator instanceof PrimitiveIterator.OfLong) {
      return (PrimitiveIterator.OfLong) iterator;
    }
    Iterable<Long> longs = () -> iterator;
    return StreamSupport.stream(longs.spliterator(), false).mapToLong(Long::valueOf).iterator();
  }
```

Comment (Stephan202) [inline guava/src/com/google/common/collect/Iterators.java]: @liach, those last two lines can be replaced with:
```java
return Streams.stream(iterator).mapToLong(Long::valueOf).iterator();
```

**Comment by cpovirk** [comment]
Bulk closing all pull requests that are listed as needing CLA signing. If you'd like us to look at your pull request, you'll need to sign the CLA and report back here.

If this is a false positive, I apologize. Please reopen the pull request, and we'll have a look.

**Pull request closed without merging**